tutorial/guias/Paso_a_paso_DIA_4.md

```diff
@@ -0,0 +1,204 @@
+
+Lo siguiente es crear las funciones que contienen las pruebas. Para la clase `Fruit` no hay mucho que probar, solamente verificaremos que la posición de la fruta sea realmente la que le pasamos: `(10, 10)`. Para ello creamos la función `test_created_as_dataclass` que no recibe parámetros. En este caso sí podemos usar el **snake_casing** porque lo único que necesitamos es que el nombre de nuestra función comience o termine con la palabra **test**. Lo que tenemos que comprobar es que el campo `position` de `fruit` sea igual a `(10, 10)`. Esto lo hacemos con la función `assertEqual` que ya está definida en la clase padre `TestCase`. De esta forma, si sucede que la igualdad no se cumple, al ejecutar nuestro test obtendremos un error. ¡Probemos si todo funciona! :grimacing:
+
+Dentro del directorio `snake` abre una terminal y ejecuta el siguiente comando:
+
+```bash
+pytest
+```
+
+Debes obtener algo similar a esto:
+
+```bash
+/snake$ pytest
+======================= test session starts ========================================
+platform linux -- Python 3.10.6, pytest-7.2.2, pluggy-1.0.0
+rootdir: /media/jj/2C3E97D93E979A82/Storage/personal-projects/codexl/snake/tutorial/DIA 4/snake
+plugins: web3-5.30.0, anyio-3.6.1
+collected 1 item
+
+tests/test_fruit.py .                                                           [100%]
+
+======================== 1 passed in 0.07s ==========================================
+```
+
+Nos va a interesar sobre todo la información a partir de la línea que empieza con *collected*. Lo que vemos en esa línea es cuántos tests se van a ejecutar, en este caso solamente 1. Luego tenemos un listado con todos los archivos que contienen tests y un punto (`.`) o una `F` para cada test en ese archivo. El punto significa que el test devolvió un resultado correcto y la `F` que el test falló.
+
+Finalmente la última línea nos da un resumen de cuántos tests pasaron, cuántos fallaron y qué tiempo tomó ejecutarlos todos. En nuestro caso, nuestro único test fue exitoso :smiley:.
+
+A modo de ejemplo, agreguemos a nuestra clase un test que sabemos que va a fallar:
+
+```python
+from unittest import TestCase
+
+from src.model.fruit import Fruit
+
+
+class TestFruit(TestCase):
+    def setUp(self):
+        self.fruit = Fruit((10, 10))
+
+    def test_created_as_dataclass(self):
+        self.assertEqual(self.fruit.position, (10, 10))
+    
+    def test_fail(self):
+        self.assertEqual(1, 2)
```

Comment:
Maybe add a comment in the code here to highlight that this is a dummy function just for showing how tests fail